reana/reana_dev/git.py

```diff
@@ -767,7 +767,23 @@ def git_upgrade(component, base):  # noqa: D301
     :type component: str
     :type base: str
     """
+
+    def _has_branch(component, branch):
```

Comment:
Perhaps we should take this function out?  It could be useful elsewhere, for example we also have in one place:

```python
if branch in get_all_branches(get_srcdir(component))
```

that could profit from the same function.  What about introducing `def branch_exists(component, branch)` next to `get_all_branches()`? 

P.S. FYI FWIW I'll be also having a PR that adds `--exclude-components` to many `git-*` functions, so that one can write things like:

```console
$ reana-dev git-status -c CLIENT -c CLUSTER --base maint-0.7 --exclude-components r-a-vomsproxy,r-a-krb5
```

Used it locally; just finishing it up everywhere...